loki/transformations/single_column/annotate.py

```diff
@@ -194,11 +195,55 @@ def transform_subroutine(self, routine, **kwargs):
             # Mark all non-parallel loops as `!$acc loop seq`
             self.annotate_sequential_loops(routine)
 
+            # Find variables with existing OpenACC data declarations
+            acc_vars = self.find_acc_vars(routine, targets)
+
             with pragmas_attached(routine, ir.Loop, attach_pragma_post=True):
-                driver_loops = find_driver_loops(routine=routine, targets=targets)
+                driver_loops = find_driver_loops(section=routine.body, targets=targets)
```

Comment:
Can't we move `find_acc_vars` inside the pragma context, to save doing this twice?

```py
with pragma_regions_attached(routine):
    with pragmas_attached(routine, ir.Loop, attach_pragma_post=True):
        acc_vars = self.find_acc_vars(routine, targets)
        driver_loops = find_driver_loops(section=routine.body, targets=targets)
        for loop in driver_loops:
         ...
```
And then remove all pragma attaching from `find_acc_vars`?

Reply:
Yes, it is much better indeed to only attach pragmas once, thanks! 🙏